src/qtcwin.c

```diff
@@ -0,0 +1,1816 @@
+				    break;
+			    case 2: activefield = 0;
+				    showfield(2);
+				    showfield(0);
+				    break;
+			    case 3: activefield = 2;
+				    showfield(2);
+				    wattrset(qtcwin, line_normal);
+				    mvwprintw(qtcwin, activefield+1, 4, "%s", qtclist.qtclines[0].qtc);
+				    break;
+			    case 4 ... 12:
+				    show_help_msg(activefield);
+				    wattrset(qtcwin, line_normal);
+				    mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[(activefield-3)].qtc);
+				    activefield--;
+				    wattrset(qtcwin, line_currinverted);
+				    mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[(activefield-3)].qtc);
+				    break;
+			}
+		    }
+		    break;
+	  case 153:		// down
+		    if (direction == RECV) {
+			if (activefield < 30) {	// last line serial field idx
+			    if (activefield == 0) {
+				activefield = 1;
+				showfield(0);
+			    }
+			    else if (activefield == 1 || activefield == 2) {
+				activefield += 2;
+				showfield(1);
+				showfield(2);
+			    }
+			    else {
+				activefield += 3;
+				showfield(activefield-3);
+			    }
+			}
+			showfield(activefield);
+			curpos = 0;
+		    }
+		    if (direction == SEND && ((trxmode != DIGIMODE && qtclist.totalsent < *qtccount) || (trxmode == DIGIMODE && qtclist.totalsent == *qtccount))) {
+			switch (activefield) {
+			    case 0: activefield = 2;
+				    showfield(0);
+				    showfield(2);
+				    break;
+			    //case 2: activefield = 3;
+				    //showfield(2);
+				    //break;
+			    case 2: activefield = 3;
+				    show_help_msg(activefield);
+				    showfield(2);
+				    wattrset(qtcwin, line_currinverted);
+				    mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[0].qtc);
+				    break;
+			    case 3 ... 11:
+				    show_help_msg(activefield);
+				    wattrset(qtcwin, line_normal);
+				    mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[(activefield-3)].qtc);
+				    activefield++;
+				    wattrset(qtcwin, line_currinverted);
+				    mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[(activefield-3)].qtc);
+				    break;
+			    case 12:
+				    activefield = 0;
+				    wattrset(qtcwin, line_normal);
+				    mvwprintw(qtcwin, 14, 4, "%s", qtclist.qtclines[9].qtc);
+				    showfield(0);
+				    break;
+			}
+		    }
+		    break;
+	  case 155:	// left
+		    if (DIRCLAUSE) {
+			if (curpos < curfieldlen) {	// curpos is a shift, means lenght - position
+			    curpos++;
+			    showfield(activefield);
+			}
+		    }
+		    break;
+	  case 154:	// right
+		    if (DIRCLAUSE) {
+			if (curpos > 0) {
+			    curpos--;
+			    showfield(activefield);
+			}
+		    }
+		    break;
+	  case 10:  		// ENTER
+		    if (activefield > 2) {
+			if (direction == RECV) {
+			    currqtc = ((activefield-3)/3);
+			    //if ((activefield-3)%3 == 2) {
+			    // TODO - is it need to restrict only for last field to allow the ENTER?
+			    if (1) {
```

Comment:
Yes, that's a question, but I don't know the answer. I don't know, what could be the good solution? There are 3 fields in a QTC line: time, callsign and serial. If the operator press the ENTER, the current QTC will saved. The question is what's the better solution? Only allow the ENTER if the current field is serial, or in any field? Now the last one is the used, but I don't know, should we change this?